nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -461,4 +482,18 @@ public DocumentRef getCommentedDocumentRef(CoreSession s, DocumentModel commentD
         });
 
     }
+
+    @Override
+    protected boolean isDocumentNotCommented(CoreSession session, DocumentModel document) {
+        String query = String.format( //
+                QUERY_GET_COMMENTS_COUNT_FOR_DOCUMENT_BY_COMMENT_ANCESTORS, document.getId());
+        return session.queryProjection(query, 0, 0).isEmpty();
```

Comment:
You should use a `limit` of `1` here, if there is one comment, the document is commented. It will avoid to query all.